ast3/Python/ast.c

```diff
@@ -1530,7 +1530,7 @@ ast_for_arguments(struct compiling *c, const node *n)
                     int res = 0;
                     i += 2; /* now follows keyword only arguments */
 
-                    if (TYPE(CHILD(n, i)) == TYPE_COMMENT) {
+                    if (i < NCH(n) && TYPE(CHILD(n, i)) == TYPE_COMMENT) {
```

Comment:
@gvanrossum: This could plausibly be the cause of `SystemError: <built-in function _parse> returned a result with an error set` with crazy line numbers.  The line number here is clearly based on uninitialized memory in the spurious case.